cli/src/ui/commands/Auth.tsx

```diff
@@ -0,0 +1,273 @@
+import React, { FC, useEffect, useState } from "react";
+import { Box, Text } from "ink";
+import Spinner from "ink-spinner";
+import fetch from "node-fetch";
+import open from "open";
+import { jwtDecode } from "jwt-decode";
+import TokenStorage from "../../utils/tokenStorage.js";
+import { getDustClient, resetDustClient } from "../../utils/dustClient.js";
+import WorkspaceSelector from "../components/WorkspaceSelector.js";
+import { ExtendedUserType } from "../../types.js";
+
+interface DeviceCodeResponse {
+  device_code: string;
+  user_code: string;
+  verification_uri: string;
+  verification_uri_complete: string;
+  expires_in: number;
+  interval: number;
+}
+
+interface TokenResponse {
+  access_token: string;
+  refresh_token: string;
+  token_type: string;
+}
+
+interface TokenErrorResponse {
+  error: string;
+  error_description?: string;
+}
+
+interface AuthProps {
+  force?: boolean;
+}
+
+const Auth: FC<AuthProps> = ({ force = false }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [deviceCode, setDeviceCode] = useState<DeviceCodeResponse | null>(null);
+  const [isPolling, setIsPolling] = useState(false);
+  const [showWorkspaceSelector, setShowWorkspaceSelector] = useState(false);
+  const [authComplete, setAuthComplete] = useState(false);
+  const [userInfo, setUserInfo] = useState<ExtendedUserType | null>(null);
+
+  const auth0Domain = process.env.AUTH0_CLIENT_DOMAIN || "";
+  const clientId = process.env.AUTH0_CLIENT_ID || "";
+  const audience = process.env.DUST_API_AUDIENCE || "";
+  const scope = "openid profile email offline_access read:user_profile";
+
+  const startDeviceFlow = async () => {
+    try {
```

Comment:
Nit: ideally we move this try catch closer to the fetch (ideally in an separate lib) and rely on Error<> 😬

Reply:
fixed